api/src/main/resources/org/openmrs/api/db/hibernate/Condition.hbm.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0"?>
+<!--
+
+    This Source Code Form is subject to the terms of the Mozilla Public License,
+    v. 2.0. If a copy of the MPL was not distributed with this file, You can
+    obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+    the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+
+    Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+    graphic logo is a trademark of OpenMRS Inc.
+
+-->
+<!DOCTYPE hibernate-mapping PUBLIC
+	"-//Hibernate/Hibernate Mapping DTD 3.1//EN"
+	"http://www.hibernate.org/dtd/hibernate-mapping-3.0.dtd">
+<hibernate-mapping package="org.openmrs">
+
+	<class name="Condition" table="condition">
+		<id name="conditionId" type="int" column="condition_id">
+			<generator class="native">
+				<param name="sequence">condition_condition_id_seq</param>
+			</generator>
+		</id>
+
+		<property name="previousVersion" type="org.openmrs.Condition" column="previous_version" />
+		<property name="onsetDate" type="java.util.Date"
+				  column="onset_date" length="19"  />
+		<property name="condition" type="java.lang.String"
```

Comment:
Is the condition supposed to be of type string?

Reply:
It is meant to be `org.openmrs.CodedOrFreeText`.

I have refactored.

Reply:
@dkayiwa 
On refactoring to `org.openmrs.CodedOrFreeText`, the application fails to compile. I am at a loss of what to do

Reply:
Am at a loss of what to say unless you give me details.

Reply:
I get this error
```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'transactionManager' defined in class path resource [applicationContext-service.xml]: Cannot resolve reference to bean 'sessionFactory' while setting bean property 'sessionFactory'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'sessionFactory' defined in class path resource [applicationContext-service.xml]: Invocation of init method failed; nested exception is org.hibernate.MappingException: Could not determine type for: org.openmrs.CodedOrFreeText, at table: condition, for columns: [org.hibernate.mapping.Column(condition)]
```

Reply:
```
org.hibernate.MappingException: Could not determine type for: 
org.openmrs.CodedOrFreeText, at table: condition, for columns: 
[org.hibernate.mapping.Column(condition)]
```

Reply:
@dkayiwa

Reply:
Ok ignore this for now and commit.

Reply:
Okay